sdcm/utils/sstable_utils.py

```diff
@@ -0,0 +1,142 @@
+    pass
+
+
+class SstableUtils:
+
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, propagation_delay_in_seconds: int, ks_cf: str = None, **kwargs):
+
+        self.db_node: BaseNode = kwargs.get("db_node", None)
+        self.db_cluster: [BaseScyllaCluster, BaseCluster] = self.db_node.parent_cluster if self.db_node else None
+        self.ks_cf = ks_cf or random.choice(self.db_cluster.get_non_system_ks_cf_list(self.db_cluster.nodes[0]))
+        self.keyspace, self.table = self.ks_cf.split('.')
+        self.propagation_delay_in_seconds = propagation_delay_in_seconds
+        self.log = logging.getLogger(self.__class__.__name__)
+        self.user = kwargs.get("user", None)
+        self.password = kwargs.get("password", None)
+
+    def count_tombstones(self):
+        sstables = self.get_sstables()
+        tombstones_num = 0
+        for sstable in sstables:
+            tombstones_num += self.count_sstable_tombstones(sstable=sstable)
+        self.log.debug('Got %s tombstones for %s', tombstones_num, self.ks_cf)
+        return tombstones_num
+
+    def get_sstables(self, from_minutes_ago: int = 0):
+        selected_sstables = []
+        ks_cf_path = self.ks_cf.replace('.', '/')
+        find_cmd = f"find /var/lib/scylla/data/{ks_cf_path}-*/*-big-Data.db -maxdepth 1 -type f"
+        if from_minutes_ago:
+            find_cmd += f" -cmin -{from_minutes_ago}"
+        try:
+            sstables_res = self.db_node.remoter.sudo(find_cmd, verbose=True)
+            if sstables_res.stderr:
+                self.log.debug('Failed to get sstables for %s. Error: %s', self.ks_cf, sstables_res.stderr)
+            else:
+                selected_sstables = sstables_res.stdout.split()
+        except Exception as error:  # pylint: disable=broad-except
+            self.log.debug('Failed to find sstables for %s: %s', self.ks_cf, error)
+
+        message = f'filtered by last {from_minutes_ago} minutes' if from_minutes_ago else '(not filtered by time)'
+        self.log.debug('Got %s sstables %s', len(selected_sstables), message)
+        return selected_sstables
+
+    def count_sstable_tombstones(self, sstable: str) -> int:
+        try:
+            self.db_node.remoter.run(f'sudo sstabledump  {sstable} 1>/tmp/sstabledump.json', verbose=False)
+            tombstones_deletion_info = self.db_node.remoter.run(
+                'sudo egrep \'"expired" : true|marked_deleted\' /tmp/sstabledump.json', verbose=False)
+            if not tombstones_deletion_info:
+                self.log.debug('Got no tombstones for sstable: %s', sstable)
+                return 0
+
+            num_tombstones = len(tombstones_deletion_info.stdout.splitlines())
+            self.log.debug('Got %s tombstones for sstable: %s', num_tombstones, sstable)
+            return num_tombstones
+        except Exception as error:  # pylint: disable=broad-except
+            self.log.error('count_sstable_tombstones failed with: %s', error)
```

Comment:
in this case, don't we want to raise an error event here?

Reply:
but, don't we expect to have tombstones?
in this exception you are catching all exceptions and writing to log instead of raising exceptions in specific cases
don't we want to catch specific failure in `grep` to return `0`, and all other cases we raise an error?

Reply:
testing if this exception can be removed.

Reply:
works ok, removed execption.